nwb_conversion_tools/utils/specdatachunkiterator.py

```diff
@@ -0,0 +1,74 @@
+"""Authors: Cody Baker, Oliver Ruebel."""
+from typing import Iterable
+import numpy as np
+
+from hdmf.data_utils import AbstractDataChunkIterator, DataChunk
+
+
+class SpecDataChunkIterator(AbstractDataChunkIterator):
+    """DataChunkIterator that lets the user specify chunk shapes."""
+
+    def __init__(self, data: Iterable, full_shape: tuple, chunk_shape: tuple, dtype: str):
+        self.data = data
+        self.full_shape = full_shape
+        self.chunk_shape = chunk_shape
+        self.__curr_timestep = 0
+        self.__curr_channel_idx = 0
+        self.__dtype = DataChunk(data=np.empty(shape=(0, 0), dtype=dtype))
+
+    def __iter__(self):
+        """Return the iterator object"""
+        return self
+
+    def __read_block(self, start_time_idx, stop_time_idx, start_ch_idx, stop_ch_idx):
```

Comment:
"time" and "channel" are specific to ElectricalSeries, and might also apply to other large arrays, but isn't as general as it could be. This might be a good stepping stone, and maybe we should continue with this for now, but ultimately I think we should be able to handle any array with an arbitrary number of dimensions (e.g. RGB video would not work in this case, since it is of shape (time, x, y, [rgb])

Reply:
Should now be generalized, but those other specific child classes still need to be made and tested.